File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/dialogs/HelpDialog.java
package ch.rmy.android.http_shortcuts.dialogs;

import android.app.Dialog;

import ch.rmy.android.http_shortcuts.utils.Destroyable;

public class HelpDialog implements Destroyable {

    private final Dialog dialog;

    protected HelpDialog(Dialog dialog) {
        this.dialog = dialog;
    }

    public void show() {
        dialog.show();
    }

    @Override
    public void destroy() {
        dialog.dismiss();
    }
}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/dialogs/HelpDialog.kt
package ch.rmy.android.http_shortcuts.dialogs

import android.app.Dialog

import ch.rmy.android.http_shortcuts.utils.Destroyable
import ch.rmy.android.http_shortcuts.utils.showIfPossible

class HelpDialog constructor(private val dialog: Dialog) {

    fun show(): Destroyable {
        dialog.showIfPossible()
        return object : Destroyable {
            override fun destroy() {
                dialog.dismiss()
            }
        }
    }

}
